packages/twenty-front/src/modules/settings/accounts/components/SettingAccountBlocklistContactRow.tsx

```diff
@@ -0,0 +1,287 @@
+      newSelectedBlocklistLevels.length === blocklistLevelDropdownItems.length
+        ? BlocklistContactLevel.ALL
+        : newSelectedBlocklistLevels.join(', '),
+    );
+  };
+
+  const handleOnDropdownClickOutside = () => {
+    setDropdownSearchText('');
+    if (isDefined(item?.id)) {
+      updateBlockedEmail({
+        id: item?.id as string,
+        handle: item?.handle as string,
+        levels: selectedBlocklistLevels,
+      } as BlocklistItem);
+    }
+  };
+
+  return (
+    <form onSubmit={submit}>
+      <StyledContainer>
+        <Dropdown
+          dropdownId={
+            !item?.id ? BLOCKLIST_CONTEXT_DROPDOWN_ID : (item?.id as string)
+          }
+          dropdownPlacement="bottom-start"
+          dropdownMenuWidth={160}
+          dropdownHotkeyScope={{ scope: BLOCKLIST_CONTEXT_DROPDOWN_ID }}
+          clickableComponent={
+            <StyledClickableComponent>
+              <StyledInputButton value={dropdownValue} placeholder="From/To" />
+              <StyledIconChevronDown />
+            </StyledClickableComponent>
+          }
+          dropdownComponents={
+            <DropdownMenuItemsContainer>
+              <StyledInput
+                value={dropdownSearchText}
+                autoFocus
+                placeholder="Search"
+                onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+                  setDropdownSearchText(event.target.value.toLowerCase());
+                }}
+              />
+              {blocklistLevelDropdownItems
+                .filter((item) =>
+                  item.label.toLowerCase().includes(dropdownSearchText),
+                )
+                .map((item) => (
+                  <MenuItemMultiSelect
+                    key={item.id}
+                    onSelectChange={() => handleMultiSelectChange(item.id)}
+                    text={item.label}
+                    selected={selectedBlocklistLevels.includes(item.id)}
+                    className={''}
+                  />
+                ))}
+            </DropdownMenuItemsContainer>
+          }
+          onClickOutside={handleOnDropdownClickOutside}
+        />
+        <StyledLinkContainer>
+          <Controller
+            name="emailOrDomain"
+            control={control}
+            render={({ field: { value, onChange }, fieldState: { error } }) => (
+              <TextInput
+                placeholder="@mydomainname.com"
+                value={!item?.id ? value : item?.handle}
+                error={error?.message}
+                fullWidth
+                readOnly={!!item?.id}
+                onChange={onChange}
+                onKeyDown={handleKeyDown}
+              />
+            )}
+          />
+        </StyledLinkContainer>
+        {item?.id ? (
+          <StyledRemoveButton
+            onClick={() => handleBlockedEmailRemove(item.id)}
+          />
+        ) : (
+          <StyledEmptyBox />
+        )}
+      </StyledContainer>
+    </form>
```

Comment:
this file is a bit long, would be nice to split it into smaller component. We could introduce a blocklist folder in settings/accounts to group all these components